nixos/modules/services/desktops/espanso.nix

```
@@ -6,19 +6,28 @@ in {
   meta = { maintainers = with lib.maintainers; [ numkem ]; };
 
   options = {
-    services.espanso = { enable = options.mkEnableOption (lib.mdDoc "Espanso"); };
+    services.espanso = {
+      enable = mkEnableOption (lib.mdDoc "Espanso");
+      package = mkPackageOption pkgs "espanso" { };
+      wayland = mkOption {
+        type = types.bool;
+        default = false;
+        description = lib.mdDoc "Use the Wayland build of Espanso.";
+      };
+    };
   };
 
   config = mkIf cfg.enable {
+    services.espanso.package = mkIf cfg.wayland pkgs.espanso-wayland;
     systemd.user.services.espanso = {
       description = "Espanso daemon";
       serviceConfig = {
-        ExecStart = "${pkgs.espanso}/bin/espanso daemon";
+        ExecStart = "${cfg.package}/bin/espanso daemon";
```

```suggestion
        ExecStart = "${lib.getExe cfg.package} daemon";
```
would be nice if that would work

---

I've added meta.mainProgram to the espanso package to make this work.